frontend/src/components/custom-tools/manage-docs-modal/ManageDocsModal.jsx

```diff
@@ -543,21 +545,32 @@ function ManageDocsModal({
         );
         updateCustomTool({ listOfDocs: newListOfDocs });
 
-        if (newListOfDocs?.length === 1 && selectedDoc?.document_id !== docId) {
-          const doc = newListOfDocs[1];
+        if (selectedDoc?.document_id === docId) {
+          const doc = newListOfDocs[0];
           handleDocChange(doc);
         }
-
-        if (docId === selectedDoc?.document_id) {
-          updateCustomTool({ selectedDoc: "" });
-          handleUpdateTool({ output: "" });
-        }
+        const updatedPromptOutput = removeIdFromCoverage(promptOutputs, docId);
+        updatePromptOutput(updatedPromptOutput);
       })
       .catch((err) => {
         setAlertDetails(handleException(err, "Failed to delete"));
       });
   };
 
+  const removeIdFromCoverage = (data, idToRemove) => {
+    return Object.entries(data).reduce((updatedData, [key, value]) => {
+      // Create a new object for the current entry
+      updatedData[key] = {
+        ...value,
+        // Update the coverage array if it exists
+        coverage: value?.coverage
+          ? value?.coverage?.filter((id) => id !== idToRemove)
+          : value?.coverage,
```

Comment:
@jagadeeswaran-zipstack The fallback to `value?.coverage` is unnecessary since filter inherently returns the same array if no changes are made. This simplifies the coverage update logic.
```suggestion
        coverage: value.coverage?.filter((id) => id !== idToRemove),
```

Reply:
Make sense, Ill make the required changes. Thanks